extensions/wikia/LookupContribs/SpecialLookupContribs_helper.php

```diff
@@ -610,163 +625,174 @@ private function getArchive( $dbr ) {
 				'ar_rev_id as rev_id',
 				'0 as rc_new',
 				'1 as page_remove'
-			),
+			],
 			$conditions,
 			__METHOD__,
-			array(
+			[
 				'LIMIT' => $this->mLimit,
 				'OFFSET' => $this->mOffset
-			)
+			]
 		);
 
 		$data['res'] = $res;
-
-		wfProfileOut( __METHOD__ );
 		return $data;
 	}
 
-	/* fetch all contributions from that given database */
-	public function fetchContribs ( ) {
-		global $wgOut, $wgRequest, $wgLang, $wgMemc, $wgContLang;
-		wfProfileIn( __METHOD__ );
+	/**
+	 * Fetch all contributions from that given database
+	 *
+	 * @return array
+	 */
+	public function fetchContribs() {
+		$memc = F::app()->wg->Memc;
 
-		$fetched_data = array(
+		$fetched_data = [
 			'cnt' => 0,
-			'data' => array()
-		);
+			'data' => []
+		];
 		$dbr = wfGetDB( DB_SLAVE, 'stats', $this->getDBname() );
```

Comment:
What about waiting to get this db handler until after we're sure the value doesn't exist in memcache?